prov/tcp/src/tcpx_ep.c

```diff
@@ -550,8 +550,8 @@ static int tcpx_ep_close(struct fid *fid)
 	tcpx_ep_tx_rx_queues_release(ep);
 	tcpx_progress_ep_del(ep);
 	ofi_close_socket(ep->conn_fd);
-	fastlock_destroy(&ep->lock);
 	ofi_endpoint_close(&ep->util_ep);
```

Comment:
@sydidelot - I wonder if we have a similar problem in ofi_endpoint_close().  The util_ep lock is destroyed at the top of that call prior to removing it from the cqs.

Reply:
@shefty I'll check out the code later today and will let you know about it.

Reply:
I looked, and I'm pretty sure that we do.  Can you move the lock destruction down in that path as well?

Reply:
good catch

Reply:
gah -- I think ofi_cq_cleanup() in  util_cq.c has the same issue wrt poll sets.  In that call fi_poll_del() needs to move up in the function.    This is apparently a good catch that may need inspection of most cleanup paths.  @a-ilango @aingerson -- please look at this fix to the tcp provider and see if your providers might have similar problems.